src/core_tests/workflow_tasks.rs

```diff
@@ -1557,3 +1557,111 @@ async fn lang_slower_than_wft_timeouts() {
     .unwrap();
     core.shutdown().await;
 }
+
+#[tokio::test]
+async fn tries_cancel_of_completed_activity() {
+    let mut t = TestHistoryBuilder::default();
+    t.add_by_type(EventType::WorkflowExecutionStarted);
+    t.add_full_wf_task();
+    let scheduled_event_id = t.add_activity_task_scheduled("1");
+    t.add_we_signaled("sig", vec![]);
```

Comment:
signal isn't really relevant to this test case, it incidentally was what unblocked the workflow in the repro.

Reply:
Yep true, I think it just makes sense to indicate something else happened that would've activated the workflow at the same time